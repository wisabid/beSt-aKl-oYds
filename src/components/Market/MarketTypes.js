import React from 'react';
import Subscription from '../../containers/Subscription';
import Outcome from '../../containers/Outcome';

const MarketTypes = (props) => {
    const { marketdata } = props;
    debugger;
    let newmarketdata = marketdata.filter(item => props.dta[0].indexOf(item.marketId) !== -1)
    return (
        <>
            <button onClick={props.goback} className="backbtn anchorl"> Home </button>
            <div className="bao-live">            
                <fieldset>
                    <legend>Category Based Market(s) Display</legend>
                    <ul className="noborder">
                        {
                            newmarketdata.map((market, indx) => {
                                    return (
                                        <>
                                        <li key={market.marketId}>
                                            <div className="inrow">
                                                <span className="anchorl"><b>{indx+1}. </b>{market.name}</span> 
                                                <span className="anchorl">( {market.type} )</span>
                                                <span className="justtitle" title="Live Price Limit">{market.liabilities.livePriceLimit} ( LPL )</span>
                                                <Subscription 
                                                    uid={`m.${market.marketId}`}
                                                />   
                                            </div>
                                            {market.name
                                                ?<Outcome eventid={market.eventId} marketid={market.marketId} />   
                                                :null                                            
                                            }                          
                                            

                                        </li>                               
                                        </>
                                    )  
                            })

                        }          
                        
                    </ul>
                </fieldset>
            </div>
        </>
    )
}

export default MarketTypes;