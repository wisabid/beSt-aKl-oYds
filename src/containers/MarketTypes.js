import { connect } from 'react-redux';
import MarketTypesComp from '../components/Market/MarketTypes';
import displayWrapper from '../utils/displayWrapper';

const mapStateToProps = (state) => {
    return {        
        marketdata : state.rL.marketdata
    }
}

const mapdispatchToProps = (dispatch) => {
    return {
        
    }
}

const MarketTypes = connect(mapStateToProps, mapdispatchToProps)(displayWrapper(MarketTypesComp));

export default MarketTypes;