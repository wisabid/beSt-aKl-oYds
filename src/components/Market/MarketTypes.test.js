import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import MarketTypes from './MarketTypes';

jest.mock('../../context/UserContext');
const mockFn = jest.fn();

describe('<MarketTypes />', () => {

  const MarketTypesComp = shallow(<MarketTypes dta={[]} goback={mockFn} marketdata={[]}/>)

  

  it('<MarketTypes /> component matches the snapshot', () => {
    expect(MarketTypesComp).toMatchSnapshot();
  })

  it('Back action click will call a function', () => {
    MarketTypesComp.find('.backbtn').simulate('click');
    expect(mockFn).toHaveBeenCalled();
  })

  

});
