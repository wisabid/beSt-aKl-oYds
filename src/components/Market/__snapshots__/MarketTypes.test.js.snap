// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MarketTypes /> <MarketTypes /> component matches the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MarketTypes
    dta={Array []}
    goback={[MockFunction]}
    marketdata={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <button
          className="backbtn anchorl"
          onClick={[MockFunction]}
        >
           Home 
        </button>,
        <div
          className="bao-live"
        >
          <fieldset>
            <legend>
              Category Based Market(s) Display
            </legend>
            <ul
              className="noborder"
            />
          </fieldset>
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": " Home ",
          "className": "backbtn anchorl",
          "onClick": [MockFunction],
        },
        "ref": null,
        "rendered": " Home ",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <fieldset>
            <legend>
              Category Based Market(s) Display
            </legend>
            <ul
              className="noborder"
            />
          </fieldset>,
          "className": "bao-live",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <legend>
                Category Based Market(s) Display
              </legend>,
              <ul
                className="noborder"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Category Based Market(s) Display",
              },
              "ref": null,
              "rendered": "Category Based Market(s) Display",
              "type": "legend",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [],
                "className": "noborder",
              },
              "ref": null,
              "rendered": Array [],
              "type": "ul",
            },
          ],
          "type": "fieldset",
        },
        "type": "div",
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <button
            className="backbtn anchorl"
            onClick={[MockFunction]}
          >
             Home 
          </button>,
          <div
            className="bao-live"
          >
            <fieldset>
              <legend>
                Category Based Market(s) Display
              </legend>
              <ul
                className="noborder"
              />
            </fieldset>
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": " Home ",
            "className": "backbtn anchorl",
            "onClick": [MockFunction],
          },
          "ref": null,
          "rendered": " Home ",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <fieldset>
              <legend>
                Category Based Market(s) Display
              </legend>
              <ul
                className="noborder"
              />
            </fieldset>,
            "className": "bao-live",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <legend>
                  Category Based Market(s) Display
                </legend>,
                <ul
                  className="noborder"
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Category Based Market(s) Display",
                },
                "ref": null,
                "rendered": "Category Based Market(s) Display",
                "type": "legend",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": Array [],
                  "className": "noborder",
                },
                "ref": null,
                "rendered": Array [],
                "type": "ul",
              },
            ],
            "type": "fieldset",
          },
          "type": "div",
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
